Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/server/BaseService.java

```diff
@@ -64,8 +66,13 @@ protected AssertionType getAssertion(WebServiceContext context, AssertionType as
         }
 
         // Extract the message id value from the WS-Addressing Header and place it in the Assertion Class
-        if (assertion != null && NullChecker.isNullish(assertion.getMessageId())) {
-            assertion.setMessageId(extractor.getOrCreateAsyncMessageId(context));
+        if (assertion != null) {
```

Comment:
[need to confirm]no need to check null b/c the obj is not null when reaching that block.